core/src/main/java/org/jdbi/v3/core/result/ResultIterable.java

```diff
@@ -76,6 +93,8 @@
     @Override
     ResultIterator<T> iterator();
 
+    StatementContext getContext();
```

Comment:
I guess I can pass them in as arguments instead.  Feels weird that we now require you to pass `(iterable, ctx)` when you get it from just so we can make the interface functional, especially when it actually does hold the context (exposed through the returned `Iterator`).